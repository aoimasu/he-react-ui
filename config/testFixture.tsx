import * as React from 'react';
import createTestContext from 'react-cosmos-test/enzyme';

type Fixture = {
  component: React.ComponentType<any>;
  name?: string;
};

function testOneFixture(fixture: Fixture, variant: string = 'a') {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component as any;
  mount();
  test(`<${displayName || name} /> rendered correctly with ${fixture.name ||
    variant} fixture`, () => expect(getWrapper()).toMatchSnapshot());
}

// TODO: use real type guard functions to discriminate one fixture vs an array of them

export default function testAllFixtures(fixtures: any, variant?: string) {
  if (!('forEach' in fixtures)) {
    testOneFixture(fixtures as Fixture, variant);
  } else if (typeof fixtures.forEach === 'function') {
    (fixtures as Fixture[]).forEach(it => testOneFixture(it));
  }
}
