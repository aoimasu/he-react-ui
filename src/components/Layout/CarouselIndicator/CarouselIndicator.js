// @flow
/*
 * CarouselIndicator
 */

// Vendor
import React from 'react';
import classnames from 'classnames';

import style from './CarouselIndicator.scss';

type Props = {
  length: number,
  current: number,
  showNext?: boolean,
  nextStep?: Function,
  className?: string,
};

function CarouselIndicator(props: Props) {
  const {
    className,
    length,
    current,
    showNext,
    nextStep,
    ...restProps
  } = props;

  return (
    <div className={classnames(style.outer, className)} {...restProps}>
      <ol className={style.wrapper}>
        {Array(length)
          .fill(1)
          .map((child, index) => (
            <li
              key={`dot-${index + 1}`}
              className={classnames(style.item, {
                [style.active]: current === index,
              })}
            />
          ))}

        {current < length - 1 && (
          <li className={style.nextBtn} onClick={nextStep}>
            Next
          </li>
        )}
      </ol>
    </div>
  );
}
CarouselIndicator.defaultProps = {
  current: 0,
  length: 0,
};
export default CarouselIndicator;
