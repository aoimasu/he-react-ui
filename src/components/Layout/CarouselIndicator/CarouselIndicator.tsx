/*
 * CarouselIndicator
 */

// Vendor
import React from 'react';
import classnames from 'classnames';

import style from './CarouselIndicator.scss';
import { returnNull } from '../../../util';

type Props = {
  length: number;
  current: number;
  nextStep?: () => void;
  onDotClick?: Function;
  className?: string;
};

const CarouselIndicator: React.SFC<Props> = props => {
  const { className, length, current, nextStep, onDotClick } = props;

  return (
    <div className={classnames(style.outer, className)}>
      <ol className={style.wrapper}>
        {Array(length)
          .fill(1)
          .map((_, index) => (
            <li
              key={`dot-${index + 1}`}
              className={classnames(style.item, {
                [style.active]: current === index,
              })}
              onClick={() => {
                if (onDotClick) onDotClick(index);
              }}
            />
          ))}

        {current < length - 1 && (
          <li className={style.nextBtn} onClick={nextStep}>
            Next
          </li>
        )}
      </ol>
    </div>
  );
};
CarouselIndicator.defaultProps = {
  current: 0,
  length: 0,
  nextStep: returnNull,
};
export default CarouselIndicator;
