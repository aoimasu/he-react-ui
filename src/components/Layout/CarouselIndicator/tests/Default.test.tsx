import { mount } from 'enzyme';
import * as React from 'react';
import testFixture from '../../../../../config/testFixture';
import CarouselIndicator from '../CarouselIndicator';
import fixture from '../fixtures/Default.fixture';

testFixture(fixture, 'Default');

test('it handles clicking the dot', () => {
  const onDotClick = jest.fn();
  const wrapper = mount(
    <CarouselIndicator length={3} current={0} onDotClick={onDotClick} />,
  );
  wrapper
    .find('li')
    .at(1)
    .simulate('click');
  expect(onDotClick).toHaveBeenCalledWith(1);
});
