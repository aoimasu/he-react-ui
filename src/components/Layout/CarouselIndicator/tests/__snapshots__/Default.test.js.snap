// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselIndicator /> rendered correctly with Default fixture 1`] = `
<CarouselIndicator
  current={1}
  length={4}
  showNext={true}
>
  <div
    className="outer"
  >
    <ol
      className="wrapper"
    >
      <li
        className="item"
        key="dot-1"
      />
      <li
        className="item active"
        key="dot-2"
      />
      <li
        className="item"
        key="dot-3"
      />
      <li
        className="item"
        key="dot-4"
      />
      <li
        className="nextBtn"
      >
        Next
      </li>
    </ol>
  </div>
</CarouselIndicator>
`;
